system/logaccess/ElasticStack/ElasticStackLogAccess.cpp

```diff
@@ -608,29 +611,58 @@ bool ElasticStackLogAccess::fetchLog(const LogAccessConditions & options, String
     return true;
 }
 
-bool ElasticStackLogStream::readLogEntries(StringBuffer & record)
+class ELASTICSTACKLOGACCESS_API ElasticStackLogStream : public CInterfaceOf<IRemoteLogAccessStream>
 {
-    Json::Value res;
-    Json::StreamWriterBuilder builder;
-    builder["indentation"] = "";
-
-    if (m_esSroller.next(res))
+public:
+    virtual bool readLogEntries(StringBuffer & record) override
     {
-        std::string str = Json::writeString(builder, res);
-        if (!res["hits"].empty())
+        Json::Value res;
+
+        if (m_esSroller.next(res))
         {
-            processESScrollJsonResp( Json::writeString(builder, res).c_str(), record, m_outputFormat);
-            return true;
+            std::string str = Json::writeString(m_jsonStreamBuilder, res);
+            if (!res["hits"].empty())
+            {
+                processESScrollJsonResp( Json::writeString(m_jsonStreamBuilder, res).c_str(), record, m_outputFormat);
+                return true;
+            }
         }
+
+        return false;
     }
 
-    return false;
+    ElasticStackLogStream(std::string & queryString, const char * connstr, const char * indexsearchpattern, LogAccessLogFormat format,  std::size_t scrollSize, std::string scrollTO)
+     : m_esSroller(std::make_shared<elasticlient::Client>(std::vector<std::string>({connstr})), scrollSize, scrollTO)
+    {
+        m_outputFormat = format;
+        m_esSroller.init(indexsearchpattern, DEFAULT_ES_DOC_TYPE, queryString);
+        m_jsonStreamBuilder.newStreamWriter();
```

Comment:
The result of this function call should be saved as a StreamWriterPtr, then needs code like:

  std::ostringstream sout;
  writer->write(value, &sout);
  process(sout.str());

**and** writer->write(res, &sout);